From 0552e4d5b0f1add075312ab2bbb2ca02bbf1790b Mon Sep 17 00:00:00 2001
From: Xiangyou Xie <xiexiangyou@huawei.com>
Date: Fri, 19 Feb 2021 15:14:32 +0800
Subject: [PATCH] arm64: Add some definitions of kvm_para*
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0552e4d5b0f1add075312ab2bbb2ca02bbf1790b


hulk inclusion
category: feature
bugzilla: 47727
CVE: NA

------------------------------

Currently, ARM does not support kvm_para* of KVM_GUEST. We provide
some definitions of kvm_para* functions, although it is only a
simple return.

Signed-off-by: Xiangyou Xie <xiexiangyou@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/include/asm/kvm_para.h | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 arch/arm64/include/asm/kvm_para.h

diff --git a/arch/arm64/include/asm/kvm_para.h b/arch/arm64/include/asm/kvm_para.h
new file mode 100644
index 000000000000..e1ecc089ee9b
--- /dev/null
+++ b/arch/arm64/include/asm/kvm_para.h
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#ifndef _ASM_ARM64_KVM_PARA_H
+#define _ASM_ARM64_KVM_PARA_H
+
+#define KVM_HINTS_REALTIME      0
+
+static inline bool kvm_check_and_clear_guest_paused(void)
+{
+	return false;
+}
+
+static inline bool kvm_para_available(void)
+{
+	return false;
+}
+
+static inline unsigned int kvm_arch_para_features(void)
+{
+	return 0;
+}
+
+static inline unsigned int kvm_arch_para_hints(void)
+{
+	return 0;
+}
+
+#endif /* _ASM_ARM64_KVM_PARA_H */
-- 
2.26.2

